[Flaky test] [extension/observer/kafkatopicsobserver] TestCollectEndpointsAllConfigSettings is flaky

### Component(s)

extension/observer/kafkatopicsobserver

### Describe the issue you're reporting

Saw the test failures in multiple PRs e.g. http://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/13527883596/job/37803241973

```
=== Failed
=== FAIL: . TestCollectEndpointsAllConfigSettings (11.00s)
    extension_test.go:112: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/observer/kafkatopicsobserver/extension_test.go:112
        	Error:      	Not equal: 
        	            	expected: []observer.Endpoint{observer.Endpoint{ID:"topic1", Target:"topic1", Details:(*observer.KafkaTopic)(0x2464d80)}, observer.Endpoint{ID:"topic2", Target:"topic2", Details:(*observer.KafkaTopic)(0x2464d80)}}
        	            	actual  : []observer.Endpoint{observer.Endpoint{ID:"topic2", Target:"topic2", Details:(*observer.KafkaTopic)(0x2464d80)}, observer.Endpoint{ID:"topic1", Target:"topic1", Details:(*observer.KafkaTopic)(0x2464d80)}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,8 @@
        	            	 ([]observer.Endpoint) (len=2) {
        	            	+ (observer.Endpoint) {
        	            	+  ID: (observer.EndpointID) (len=6) "topic2",
        	            	+  Target: (string) (len=6) "topic2",
        	            	+  Details: (*observer.KafkaTopic)({
        	            	+  })
        	            	+ },
        	            	  (observer.Endpoint) {
        	            	@@ -6,8 +12,2 @@
        	            	   })
        	            	- },
        	            	- (observer.Endpoint) {
        	            	-  ID: (observer.EndpointID) (len=6) "topic2",
        	            	-  Target: (string) (len=6) "topic2",
        	            	-  Details: (*observer.KafkaTopic)({
        	            	-  })
        	            	  }
        	Test:       	TestCollectEndpointsAllConfigSettings

=== FAIL: . TestCollectEndpointsAllConfigSettings (re-run 1) (6.00s)
    extension_test.go:108: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/extension/observer/kafkatopicsobserver/extension_test.go:108
        	Error:      	Not equal: 
        	            	expected: []observer.Endpoint{observer.Endpoint{ID:"topic1", Target:"topic1", Details:(*observer.KafkaTopic)(0x2464d80)}, observer.Endpoint{ID:"topic2", Target:"topic2", Details:(*observer.KafkaTopic)(0x2464d80)}}
make[2]: *** [../../../Makefile.Common:102: test] Error 1
make[1]: *** [Makefile:231: extension/observer/kafkatopicsobserver] Error 2
make: *** [Makefile:149: gotest] Error 2
        	            	actual  : []observer.Endpoint{observer.Endpoint{ID:"topic2", Target:"topic2", Details:(*observer.KafkaTopic)(0x2464d80)}, observer.Endpoint{ID:"topic1", Target:"topic1", Details:(*observer.KafkaTopic)(0x2464d80)}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,8 @@
        	            	 ([]observer.Endpoint) (len=2) {
        	            	+ (observer.Endpoint) {
        	            	+  ID: (observer.EndpointID) (len=6) "topic2",
        	            	+  Target: (string) (len=6) "topic2",
        	            	+  Details: (*observer.KafkaTopic)({
        	            	+  })
        	            	+ },
        	            	  (observer.Endpoint) {
        	            	@@ -6,8 +12,2 @@
        	            	   })
        	            	- },
        	            	- (observer.Endpoint) {
        	            	-  ID: (observer.EndpointID) (len=6) "topic2",
        	            	-  Target: (string) (len=6) "topic2",
        	            	-  Details: (*observer.KafkaTopic)({
        	            	-  })
        	            	  }
        	Test:       	TestCollectEndpointsAllConfigSettings

DONE 2 runs, 9 tests, 2 failures in 26.199s
```

--- a/extension/observer/kafkatopicsobserver/extension_test.go
+++ b/extension/observer/kafkatopicsobserver/extension_test.go
@@ -105,11 +105,11 @@ func TestCollectEndpointsAllConfigSettings(t *testing.T) {
 			Details: &observer.KafkaTopic{},
 		},
 	}
-	require.Equal(t, want, kEndpoints)
+	require.ElementsMatch(t, want, kEndpoints)
 
 	time.Sleep(5 * time.Second)
 	kEndpoints = obvs.ListEndpoints()
-	require.Equal(t, want, kEndpoints)
+	require.ElementsMatch(t, want, kEndpoints)
 
 	time.Sleep(5 * time.Second)
 	kEndpoints = obvs.ListEndpoints()
@@ -120,7 +120,7 @@ func TestCollectEndpointsAllConfigSettings(t *testing.T) {
 			Details: &observer.KafkaTopic{},
 		},
 	}
-	require.Equal(t, want, kEndpoints)
+	require.ElementsMatch(t, want, kEndpoints)
 
 	err = obvs.Shutdown(context.Background())
 	require.NoError(t, err)

